zebrad/src/commands/entry_point.rs

```diff
@@ -0,0 +1,138 @@
+//! Zebrad EntryPoint
+
+use abscissa_core::{Command, Configurable, FrameworkError, Runnable};
+use clap::Parser;
+use std::{ffi::OsString, path::PathBuf};
+
+use crate::config::ZebradConfig;
+
+use super::ZebradCmd;
+
+/// Toplevel entrypoint command.
+///
+/// Handles obtaining toplevel help as well as verbosity settings.
+#[derive(Debug, clap::Parser)]
+#[clap(
+    version = clap::crate_version!(),
+    author="Zcash Foundation <zebra@zfnd.org>",
+    help_template = "\
+{name} {version}\n
+{author}\n
+{usage-heading} {usage}\n
+{all-args}\
+"
+)]
+pub struct EntryPoint {
+    /// Subcommand to execute.
+    ///
+    /// The `command` option will delegate option parsing to the command type,
+    /// starting at the first free argument. Defaults to start.
+    #[clap(subcommand)]
+    pub cmd: Option<ZebradCmd>,
+
+    /// Path to the configuration file
+    #[clap(long, short, help = "path to configuration file")]
+    pub config: Option<PathBuf>,
+
+    /// Increase verbosity setting
+    #[clap(long, short, help = "be verbose")]
+    pub verbose: bool,
+
+    /// Filter strings which override the config file and defaults
+    // This can be applied to the default start command if no subcommand is provided.
+    #[clap(help = "tracing filters which override the zebrad.toml config")]
+    filters: Vec<String>,
+}
+
+impl EntryPoint {
+    /// Borrow the command in the option
+    ///
+    /// # Panics
+    ///
+    /// If `cmd` is None
+    pub fn cmd(&self) -> &ZebradCmd {
+        self.cmd
+            .as_ref()
+            .expect("should default to start if not provided")
+    }
+
+    /// Returns a string that parses to the default subcommand
+    pub fn default_cmd_as_str() -> &'static str {
+        "start"
+    }
+
+    /// Process command arguments and insert the default subcommand
+    /// if no subcommand is provided.
+    pub fn process_cli_args(args: Vec<OsString>) -> Vec<OsString> {
```

Comment:
Required for the simplification below

```suggestion
    pub fn process_cli_args(mut args: Vec<OsString>) -> Vec<OsString> {
```